src/App.tsx

```diff
@@ -40,6 +50,19 @@ export const App = () => {
     (state) => state.notification.notifications
   );
 
+  // handle dark mode system change
+  useEffect(() => {
```

Comment:
This is the opposite flow of what we have,
we use user input to change the theme and the LS and in here you added a change to LS and theme after changing the theme